app/models/product_import/entry_validator.rb

```diff
@@ -189,6 +196,43 @@ def unit_fields_validation(entry)
                              error: I18n.t('admin.product_import.model.conditional_blank'))
     end
 
+    def is_numeric(value)
+      return true unless Float(value, exception: false).nil?
+    end
+
+    def price_validation(entry)
+      return if is_numeric(entry.price)
+
+      if empty_or_placeholder_value(entry.price)
+        mark_as_invalid(entry, attribute: 'price',
+                               error: I18n.t('admin.product_import.model.blank'))
+      else
+        mark_as_invalid(entry, attribute: 'price',
+                               error: I18n.t('admin.product_import.model.incorrect_value'))
+      end
+    end
+
+    def on_hand_on_demand_validation(entry)
+      on_hand_present_numeric = !empty_or_placeholder_value(entry.on_hand) &&
+                                is_numeric(entry.on_hand)
+      on_hand_value = entry.on_hand&.to_i
+      on_demand_present_numeric = !empty_or_placeholder_value(entry.on_demand) &&
+                                  is_numeric(entry.on_demand)
+      on_demand_value = entry.on_demand&.to_i
+
+      return if (on_hand_present_numeric && on_hand_value >= 0) ||
+                (on_demand_present_numeric && on_demand_value == 1)
+
+      mark_as_invalid(entry, attribute: 'on_hand',
+                             error: I18n.t('admin.product_import.model.incorrect_value'))
+      mark_as_invalid(entry, attribute: 'on_demand',
+                             error: I18n.t('admin.product_import.model.incorrect_value'))
+    end
+
+    def empty_or_placeholder_value(value)
+      !value&.present? || value.nil? || value.to_s.strip == '' || value.to_s.strip == "-"
```

Comment:
I think the first three conditions are equivalent to `#blank?` (https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html ). Would you be able to try that?

Reply:
You are correct.
Thank you. Will make the suggested changes.